background.tex

```diff
@@ -3,62 +3,63 @@
+In the present study, we identify the cause of the straggler tasks in the context of analyzing large MD trajectories and provide solutions through which we can improve performance and scaling.
+Even though the proposed solutions to avoid stragglers are specifically applied to the analysis of MD trajectories in \package{MDAnalysis}, all of these principles and implementation techniques are potentially applicable to data analysis in other Python-based libraries.
 
 \subsection{Other packages with parallel analysis capabilities}
 \label{sec:otherparallel}
 
 There have been various attempts to parallelize the analysis of MD trajectories. 
-HiMach \cite{himach-2008} introduces scalable and flexible parallel Python framework to deal with massive MD trajectories, by combining and extending the strengths of Google's MapReduce and the VMD analysis tool \cite{Hum96}. 
-HiMach's runtime is responsible to parallelize and distribute the Map and Reduce classes to the assigned cores.
-HiMach uses parallel I/O for file access during the map tasks and \text{MPI\_Allgather} in the reduction process. 
-HiMach, however, does not discuss parallel analysis for the task types which can not be analyzed using a MapReduce implementation.
-The analysis needs by MD community, however, tend to be varied and a generalized parallel framework should be able to provide parallel analysis for a variety of task types.
-Furthermore, HiMach is not available under an open source license, which makes it difficult to improve upon this work or integrate it in more complex analysis workflows.
-
-Wu et. al. \cite{Wu_et.al} present a scalable parallel framework for distributed-memory post-simulation data analysis.
+HiMach~\cite{himach-2008} introduces scalable and flexible parallel Python framework to deal with massive MD trajectories, by combining and extending Google's MapReduce and the VMD analysis tool~\cite{Hum96}. 
+HiMach's runtime is responsible to parallelize and distribute Map and Reduce classes to assigned cores.
+HiMach uses parallel I/O for file access during map tasks and \text{MPI\_Allgather} in the reduction process. 
+HiMach, however, does not discuss parallel analysis of analysis types that cannot be implemented via MapReduce.
+\gpnote{I do not understand this sentence at all. I do not think it is needed at all here. I suggest to remove it. :
+The analysis needs by MD community, however, tend to be varied and a generalized parallel framework should be able to provide parallel analysis for a variety of task types.}
+\gpnote{ Arguing about open source does not have to do with the problem this paper addresses. I suggest to remove: Furthermore, HiMach is not available under an open source license, which does not allow community contributions and continuous integration with state-of-the-art methods.}
```

Comment:
This is also out-of-scope in my opinion. Regardless if a framework is open source or not it may offer parallelization to MD analysis algorithms.

Reply:
Yes, but that is a limitation right? And we are arguing the pros and cons of previous works.

Reply:
Yes, we are arguing about the pros and cons of related with our work. An argument for closed source software is as valid as an argument against here. I am not sure it adds to what the paper is trying to say.